## Estadistica Aplicada
Actividad 6

Jesus Adrian Cordero Hernandez 1952778

Gpo 41

# *1. Importa las librerías básicas y especializadas que consideres necesarias* 

In [203]:
## Librerias basicas
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [204]:
## Librerias especializadas
from lifelines import CoxPHFitter
import seaborn as sns
from lifelines import ExponentialFitter
from lifelines import WeibullFitter
from lifelines import LogNormalFitter

# *2. Importa la base de datos*

In [205]:
datos = pd.read_csv('https://raw.githubusercontent.com/jimmyzac/Estadistica-Aplicada-FCFM-UANL/main/bases_datos/desercion_laboral.csv')
datos

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0               3.8         5.3          2                 157         3   
1               8.0         8.6          5                 262         6   
2               1.1         8.8          7                 272         4   
3               7.2         8.7          5                 223         5   
4               3.7         5.2          2                 159         3   
...             ...         ...        ...                 ...       ...   
14994           4.0         5.7          2                 151         3   
14995           3.7         4.8          2                 160         3   
14996           3.7         5.3          2                 143         3   
14997           1.1         9.6          6                 280         4   
14998           3.7         5.2          2                 158         3   

       accidente  renunciar  ascenso_5years departamento salario  
0              0          1               0       ventas    bajo  
1              0          1               0       ventas   medio  
2              0          1               0       ventas   medio  
3              0          1               0       ventas    bajo  
4              0          1               0       ventas    bajo  
...          ...        ...             ...          ...     ...  
14994          0          1               0      soporte    bajo  
14995          0          1               0      soporte    bajo  
14996          0          1               0      soporte    bajo  
14997          0          1               0      soporte    bajo  
14998          0          1               0      soporte    bajo  

[14999 rows x 10 columns]

# *3. Depurar la base de datos y estadísticas descriptivas*

In [206]:
## 3.a) Verificar que no haya missing values, en caso de haberlos decidir imputar o eliminar
datos.isnull().sum()
datos = datos.dropna()
datos.isnull().sum()

satisfaccion          0
evaluacion            0
proyectos             0
promedio_horas_mes    0
semestre              0
accidente             0
renunciar             0
ascenso_5years        0
departamento          0
salario               0
dtype: int64

In [207]:
## 3.b) Verificar que no haya valores duplicados con dataframe.duplicated().sum()en caso de haber valores duplicados eliminarlos con dataframe=dataframe.drop_duplicates(keep='first').reset_index(drop=True)
datos.duplicated().sum()
datos = datos.drop_duplicates(keep='first').reset_index(drop=True)
datos.duplicated().sum()

0

In [208]:
## 3.c) Verificar  que  todas  las  variables  sean  numéricas,  si  no  son  numéricas hacerlas   numéricas,   en   el   caso   de   las   categóricas   convertirlas   en dummies y conservar n-1 dummies
datos.dtypes
datos['satisfaccion'] = datos['satisfaccion'].astype('int64')
datos['evaluacion'] = datos['evaluacion'].astype('int64')
datos.dtypes

satisfaccion           int64
evaluacion             int64
proyectos              int64
promedio_horas_mes     int64
semestre               int64
accidente              int64
renunciar              int64
ascenso_5years         int64
departamento          object
salario               object
dtype: object

In [209]:
dummies = pd.get_dummies(datos['departamento'], prefix='dep').astype(int)
dummies

dep_Investigación  dep_administración  dep_contabilidad  \
0                      0                   0                 0   
1                      0                   0                 0   
2                      0                   0                 0   
3                      0                   0                 0   
4                      0                   0                 0   
...                  ...                 ...               ...   
11986                  0                   0                 0   
11987                  0                   0                 0   
11988                  0                   0                 0   
11989                  0                   0                 0   
11990                  0                   0                 0   

       dep_desarrollo de producto  dep_informática  dep_mantenimiento  \
0                               0                0                  0   
1                               0                0                  0   
2                               0                0                  0   
3                               0                0                  0   
4                               0                0                  0   
...                           ...              ...                ...   
11986                           1                0                  0   
11987                           1                0                  0   
11988                           1                0                  0   
11989                           1                0                  0   
11990                           0                1                  0   

       dep_mercadotecnia  dep_recursos humanos  dep_soporte  dep_ventas  
0                      0                     0            0           1  
1                      0                     0            0           1  
2                      0                     0            0           1  
3                      0                     0            0           1  
4                      0                     0            0           1  
...                  ...                   ...          ...         ...  
11986                  0                     0            0           0  
11987                  0                     0            0           0  
11988                  0                     0            0           0  
11989                  0                     0            0           0  
11990                  0                     0            0           0  

[11991 rows x 10 columns]

In [210]:
dummies = dummies[['dep_administración','dep_contabilidad','dep_desarrollo de producto','dep_informática','dep_mantenimiento','dep_mercadotecnia','dep_recursos humanos','dep_soporte','dep_ventas']]
datos = pd.concat([datos, dummies], axis=1)
datos = datos.drop('departamento', axis=1)
datos.dtypes
datos.isnull().sum()

satisfaccion                  0
evaluacion                    0
proyectos                     0
promedio_horas_mes            0
semestre                      0
accidente                     0
renunciar                     0
ascenso_5years                0
salario                       0
dep_administración            0
dep_contabilidad              0
dep_desarrollo de producto    0
dep_informática               0
dep_mantenimiento             0
dep_mercadotecnia             0
dep_recursos humanos          0
dep_soporte                   0
dep_ventas                    0
dtype: int64

In [211]:
dummies2 = pd.get_dummies(datos['salario'], prefix='sal').astype(int)
dummies2

sal_alto  sal_bajo  sal_medio
0             0         1          0
1             0         0          1
2             0         0          1
3             0         1          0
4             0         1          0
...         ...       ...        ...
11986         0         1          0
11987         0         1          0
11988         0         1          0
11989         0         1          0
11990         0         1          0

[11991 rows x 3 columns]

In [212]:
dummies2 = dummies2[['sal_bajo','sal_medio']]
datos = pd.concat([datos, dummies2], axis=1)
datos = datos.drop('salario', axis=1)
datos.dtypes
datos.isnull().sum()

satisfaccion                  0
evaluacion                    0
proyectos                     0
promedio_horas_mes            0
semestre                      0
accidente                     0
renunciar                     0
ascenso_5years                0
dep_administración            0
dep_contabilidad              0
dep_desarrollo de producto    0
dep_informática               0
dep_mantenimiento             0
dep_mercadotecnia             0
dep_recursos humanos          0
dep_soporte                   0
dep_ventas                    0
sal_bajo                      0
sal_medio                     0
dtype: int64

In [213]:
## 3.d) Calcular estadísticas descriptivas
datos.describe()

satisfaccion    evaluacion     proyectos  promedio_horas_mes  \
count  11991.000000  11991.000000  11991.000000        11991.000000   
mean       5.851889      6.707948      3.802852          200.473522   
std        2.397506      1.703189      1.163238           48.727813   
min        0.000000      3.000000      2.000000           96.000000   
25%        4.000000      5.000000      3.000000          157.000000   
50%        6.000000      7.000000      4.000000          200.000000   
75%        8.000000      8.000000      5.000000          243.000000   
max       10.000000     10.000000      7.000000          310.000000   

           semestre     accidente     renunciar  ascenso_5years  \
count  11991.000000  11991.000000  11991.000000    11991.000000   
mean       3.364857      0.154282      0.166041        0.016929   
std        1.330240      0.361234      0.372133        0.129012   
min        2.000000      0.000000      0.000000        0.000000   
25%        3.000000      0.000000      0.000000        0.000000   
50%        3.000000      0.000000      0.000000        0.000000   
75%        4.000000      0.000000      0.000000        0.000000   
max       10.000000      1.000000      1.000000        1.000000   

       dep_administración  dep_contabilidad  dep_desarrollo de producto  \
count        11991.000000      11991.000000                11991.000000   
mean             0.036361          0.051789                    0.057210   
std              0.187194          0.221610                    0.232252   
min              0.000000          0.000000                    0.000000   
25%              0.000000          0.000000                    0.000000   
50%              0.000000          0.000000                    0.000000   
75%              0.000000          0.000000                    0.000000   
max              1.000000          1.000000                    1.000000   

       dep_informática  dep_mantenimiento  dep_mercadotecnia  \
count     11991.000000        11991.00000       11991.000000   
mean          0.081394            0.18714           0.056125   
std           0.273451            0.39004           0.230173   
min           0.000000            0.00000           0.000000   
25%           0.000000            0.00000           0.000000   
50%           0.000000            0.00000           0.000000   
75%           0.000000            0.00000           0.000000   
max           1.000000            1.00000           1.000000   

       dep_recursos humanos   dep_soporte    dep_ventas      sal_bajo  \
count          11991.000000  11991.000000  11991.000000  11991.000000   
mean               0.050121      0.151864      0.270119      0.478692   
std                0.218204      0.358904      0.444040      0.499567   
min                0.000000      0.000000      0.000000      0.000000   
25%                0.000000      0.000000      0.000000      0.000000   
50%                0.000000      0.000000      0.000000      0.000000   
75%                0.000000      0.000000      1.000000      1.000000   
max                1.000000      1.000000      1.000000      1.000000   

          sal_medio  
count  11991.000000  
mean       0.438746  
std        0.496254  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

Que tan satisfecho se encuentra la persona con su trabajo en una escala de 0 a 10 es de 5.85, Que tan bien realiza su trabajo la persona en una escala de 0 a 10 es de 6.7, Número de proyectos asignados al empleado es de 3.8, Promedio de horas mensuales trabajadas es de 200.47h, Semestres que ha trabajado en la empresa es de 3.36, casi no hubo accidentes con 15.42% de ocurrencia, casi no hubo ascensos con una probabilidad de 1.69% de ascendidos, el mayor porcentaje de los asalareados se encuentra en la media y hay mas trabajadores en el departamento de ventas

# *4. Estimar el modelode Cox e interpretar cada uno de los coeficientes*

In [214]:
cph1 = CoxPHFitter().fit(datos, 'semestre', 'renunciar')
cph1.print_summary()

<lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
      baseline estimation = breslow
   number of observations = 11991
number of events observed = 1991
   partial log-likelihood = -15905.76
         time fit was run = 2023-09-26 23:06:53 UTC

---
                             coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
covariate                                                                                                                          
satisfaccion                -0.24       0.79       0.01            -0.25            -0.22                 0.78                 0.80
evaluacion                  -0.01       0.99       0.01            -0.04             0.01                 0.96                 1.01
proyectos                   -0.28       0.75       0.02            -0.32            -0.24                 0.73                 0.78
promedio_horas_mes           0.00       1.00       0.00             0.00             0.00                 1.00                 1.00
accidente                   -1.24       0.29       0.10            -1.44            -1.05                 0.24                 0.35
ascenso_5years              -1.37       0.25       0.36            -2.07            -0.67                 0.13                 0.51
dep_administración           0.23       1.25       0.18            -0.12             0.57                 0.89                 1.77
dep_contabilidad             0.26       1.30       0.14            -0.02             0.54                 0.98                 1.72
dep_desarrollo de producto   0.35       1.42       0.14             0.07             0.64                 1.07                 1.89
dep_informática              0.30       1.35       0.13             0.04             0.56                 1.04                 1.76
dep_mantenimiento            0.44       1.55       0.12             0.20             0.67                 1.23                 1.96
dep_mercadotecnia            0.35       1.41       0.14             0.06             0.63                 1.07                 1.87
dep_recursos humanos         0.45       1.57       0.14             0.17             0.73                 1.18                 2.08
dep_soporte                  0.46       1.58       0.12             0.22             0.70                 1.24                 2.01
dep_ventas                   0.36       1.43       0.12             0.13             0.59                 1.14                 1.80
sal_bajo                     1.54       4.65       0.15             1.25             1.83                 3.48                 6.22
sal_medio                    1.15       3.15       0.15             0.85             1.44                 2.35                 4.22

                             cmp to      z      p   -log2(p)
covariate                                                   
satisfaccion                   0.00 -26.22 <0.005     500.95
evaluacion                     0.00  -0.97   0.33       1.60
proyectos                      0.00 -14.30 <0.005     151.75
promedio_horas_mes             0.00   4.10 <0.005      14.55
accidente                      0.00 -12.39 <0.005     114.67
ascenso_5years                 0.00  -3.85 <0.005      13.08
dep_administración             0.00   1.27   0.20       2.30
dep_contabilidad               0.00   1.79   0.07       3.78
dep_desarrollo de producto     0.00   2.43   0.01       6.06
dep_informática                0.00   2.23   0.03       5.28
dep_mantenimiento              0.00   3.67 <0.005      11.98
dep_mercadotecnia              0.00   2.40   0.02       5.93
dep_recursos humanos           0.00   3.14 <0.005       9.19
dep_soporte                    0.00   3.73 <0.005      12.37
dep_ventas                     0.00   3.07 <0.005       8.87
sal_bajo                       0.00  10.35 <0.005      80.94
sal_medio          

satisfacción, se rechaza H0, que un empleado este satisfecho reduce en un 21% el riesgo de que renuncie de la empresa.

evaluación, No se rechaza H0, la evalaución no afecta el riesgo de que un empleado renuncie.

proyectos, se rechaza H0, el número de proyectos reduce en un 25% el riesgo de que un empleado renuncie.

promedio_horas_mes, se rechaza H0, el número de horas trabajadas aumenta en un 0.19% el riesgo de que un empleado renuncie.

accidentes, se rechaza H0, si tuvieron accidentes reduce en un 71% el riesgo de que un empleado renuncie con respecto con los que no tuvieron un accidente.

ascenso_5years, se rechaza H0, si un empleado tiene un ascenso reduce en un 75% el riesgo de que renuncie con respecto a los que no tuvieron un ascenso.

departamento de administración,No se rechaza Ho, trabajar en el departamento de adminstración no afecta al riesgo de renuncia con respecto a trabajar en el departamento de investigación.

departamento de contabilidad,No se rechaza Ho, trabajar en el departamento de contabilidad no afecta al riesgo de renuncia con respecto a trabajar en el departamento de investigación.

departamento de desarrollo de productos,Se rechaza Ho, trabajar en el departamento de desarrollo de productos incrementa en un 42% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de informática,Se rechaza Ho, trabajar en el departamento de informática incrementa en un 35% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de mantenimiento,Se rechaza Ho, trabajar en el departamento de mantenimiento incrementa en un 55% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de mercadotecnia,Se rechaza Ho, trabajar en el departamento de mercadotecnia incrementa en un 41% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de recursos humanos,Se rechaza Ho, trabajar en el departamento de recursos humanos incrementa en un 57% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de soporte,Se rechaza Ho, trabajar en el departamento de soporte incrementa en un 58% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

departamento de ventas,Se rechaza Ho, trabajar en el departamento de ventas incrementa en un 43% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

salario bajo,Se rechaza H0, recibir un salario bajo incrementa en un 365% el riesgo de que un empleado renuncie con respecto a los que ganan un salrio alto.

salario medio, Se rechaza H0, recibir un salario medio incrementa el riesgo de que un empleado renuncie en un 215% con respecto a los que ganan un salario alto.

# *5. Con base  en la interpretación de los coeficientes,¿qué recomendación podría hacerle  a  la  empresa  para  que  disminuya  la  cantidad  de  renuncias  de  los empleados?*

Lo que puede hacer la empresa para disminuir las renuncias es mantener a los empelados satisfechos, asignarles una cantidad prudente de proyectos, no ponerlos a trabajar un largo horario y darles ascenso en un perido no mayor de 5 años.

# *6. Pruebe si el modelo de Cox cumple el supuesto de riesgos proporcionales, de no hacerlo estime el modelo AFT adecuado*

In [215]:
cph1.check_assumptions(datos, p_value_threshold=0.05)
## H0: se viola el supuesto
## H1: se cumple

The ``p_value_threshold`` is set at 0.05. Even under the null hypothesis of no violations, some
covariates will be below the threshold by chance. This is compounded when there are many covariates.
Similarly, when there are lots of observations, even minor deviances from the proportional hazard
assumption will be flagged.

With that in mind, it's best to use a combination of statistical tests and visual tests to determine
the most serious violations. Produce visual plots using ``check_assumptions(..., show_plots=True)``
and looking for non-constant lines. See link [A] below for a full example.



<lifelines.StatisticalResult: proportional_hazard_test>
 null_distribution = chi squared
degrees_of_freedom = 1
             model = <lifelines.CoxPHFitter: fitted with 11991 total observations, 10000 right-censored observations>
         test_name = proportional_hazard_test

---
                                 test_statistic      p  -log2(p)
accidente                  km              0.06   0.80      0.32
                           rank            1.33   0.25      2.00
ascenso_5years             km              3.93   0.05      4.40
                           rank            7.11   0.01      7.03
dep_administración         km              5.13   0.02      5.41
                           rank            5.76   0.02      5.93
dep_contabilidad           km              0.23   0.63      0.66
                           rank            1.51   0.22      2.19
dep_desarrollo de producto km              0.83   0.36      1.46
                           rank            0.72   0.39      1.34
dep_informática            km              0.61   0.43      1.21
                           rank            0.67   0.41      1.28
dep_mantenimiento          km              0.60   0.44      1.19
                           rank            2.50   0.11      3.14
dep_mercadotecnia          km              2.64   0.10      3.26
                           rank            6.26   0.01      6.34
dep_recursos humanos       km              0.96   0.33      1.61
                           rank            3.36   0.07      3.90
dep_soporte                km              0.01   0.92      0.12
                           rank            1.01   0.32      1.66
dep_ventas                 km              1.28   0.26      1.96
                           rank            4.82   0.03      5.15
evaluacion                 km            456.70 <0.005    334.19
                           rank          433.86 <0.005    317.68
promedio_horas_mes         km            262.79 <0.005    193.91
                           rank          303.15 <0.005    223.13
proyectos                  km            564.24 <0.005    411.91
                           rank          685.75 <0.005    499.70
sal_bajo                   km              0.36   0.55      0.87
                           rank            0.00   0.96      0.06
sal_medio                  km              0.15   0.69      0.53
                           rank            0.10   0.75      0.42
satisfaccion               km           1217.52 <0.005    883.70
                           rank          825.64 <0.005    600.75



1. Variable 'satisfaccion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'satisfaccion' might be incorrect. That is, there
may be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to specify a functional form.

   Advice 2: try binning the variable 'satisfaccion' using pd.cut, and then specify it in
`strata=['satisfaccion', ...]` in the call in `.fit`. See documentation in link [B] below.

   Advice 3: try adding an interaction term with your time variable. See documentation in link [C]
below.


2. Variable 'evaluacion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'evaluacion' might be incorrect. That is, there may
be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to 

[]

Como no se cumple el supuesto debemos utilizar un modelo alternativo, El modelo de riesgo acelerado AFT.

Primero debemos identificar si es exponencial, weibull o lognormal

In [216]:
mexpo = ExponentialFitter().fit(datos['semestre'],datos['renunciar'])
mweibull = WeibullFitter().fit(datos['semestre'],datos['renunciar'])
mlogn = LogNormalFitter().fit(datos['semestre'],datos['renunciar'])
mexpo.AIC_, mweibull.AIC_, mlogn.AIC_

(15965.45892201228, 12511.512867661068, 11587.92169210356)

In [217]:
## vamos a elegir el modelo LogNormal
from lifelines import LogNormalAFTFitter
lognAFT = LogNormalAFTFitter().fit(datos,'semestre','renunciar')
lognAFT.print_summary()

<lifelines.LogNormalAFTFitter: fitted with 11991 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
   number of observations = 11991
number of events observed = 1991
           log-likelihood = -4899.95
         time fit was run = 2023-09-26 23:06:57 UTC

---
                                    coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
param  covariate                                                                                                                          
mu_    accidente                    0.26       1.30       0.02             0.22             0.30                 1.25                 1.35
       ascenso_5years               0.36       1.44       0.06             0.24             0.49                 1.27                 1.63
       dep_administración           0.05       1.05       0.04            -0.03             0.13                 0.97                 1.14
       dep_contabilidad            -0.05       0.96       0.03            -0.11             0.02                 0.89                 1.02
       dep_desarrollo de producto  -0.05       0.95       0.03            -0.12             0.02                 0.89                 1.02
       dep_informática             -0.05       0.95       0.03            -0.11             0.01                 0.90                 1.01
       dep_mantenimiento           -0.08       0.92       0.03            -0.13            -0.02                 0.88                 0.98
       dep_mercadotecnia           -0.04       0.96       0.03            -0.10             0.03                 0.90                 1.03
       dep_recursos humanos        -0.09       0.92       0.03            -0.16            -0.02                 0.86                 0.98
       dep_soporte                 -0.08       0.92       0.03            -0.14            -0.03                 0.87                 0.97
       dep_ventas                  -0.06       0.94       0.03            -0.11            -0.00                 0.90                 1.00
       evaluacion                   0.01       1.01       0.00            -0.00             0.01                 1.00                 1.01
       promedio_horas_mes          -0.00       1.00       0.00            -0.00            -0.00                 1.00                 1.00
       proyectos                    0.09       1.10       0.01             0.08             0.10                 1.09                 1.11
       sal_bajo                    -0.37       0.69       0.03            -0.43            -0.31                 0.65                 0.73
       sal_medio                   -0.29       0.75       0.03            -0.35            -0.23                 0.71                 0.80
       satisfaccion                 0.07       1.07       0.00             0.06             0.07                 1.06                 1.07
       Intercept                    1.39       4.01       0.04             1.30             1.48                 3.68                 4.38
sigma_ Intercept                   -1.11       0.33       0.02            -1.14            -1.08                 0.32                 0.34

                                    cmp to      z      p   -log2(p)
param  covariate                                                   
mu_    accidente                      0.00  12.95 <0.005     124.97
       ascenso_5years                 0.00   5.74 <0.005      26.69
       dep_administración             0.00   1.26   0.21       2.26
       dep_contabilidad               0.00  -1.34   0.18       2.48
       dep_desarrollo de producto     0.00  -1.46   0.14       2.80
       dep_informática                0.00  -1.51   0.13       2.92
       dep_mantenimiento              0.00  -2.81 <0.005       7.65
       dep_mercadotecnia              0.00  -1.07   0.28       1.81
       dep_recursos humanos           0.00  -2.60   0.01

# *7. Interprete los coeficientes del modelo AFT*

accidentes, se rechaza H0, si tuvieron accidentes reduce en un 30% el riesgo de que un empleado renuncie con respecto con los que no tuvieron un accidente.

ascenso_5years, se rechaza H0, si un empleado tiene un ascenso, reduce en un 44% el riesgo de que renuncie con respecto a los que no tuvieron un ascenso.

dep_administracion, No se rechaza Ho, trabajar en el departamento de adminstración no afecta al riesgo de renuncia con respecto a trabajar en el departamento de investigación.

departamento de contabilidad, No se rechaza Ho, trabajar en el departamento de contabilidad no afecta al riesgo de renuncia con respecto a trabajar en el departamento de investigación.

dep_desarrollo de producto, No se rechaza Ho, trabajar en el departamento de desarrollo de productos no afecta el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_informatica, No se rechaza Ho, trabajar en el departamento de informática no afecta el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_mantenimiento, Se rechaza Ho, trabajar en el departamento de mantenimiento incrementa en un 8% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_mercadoctenia, No se rechaza Ho, trabajar en el departamento de mercadotecnia no afecta el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_recursos humanos,No se rechaza Ho, trabajar en el departamento de recursos humanos no afecta el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_soporte, Se rechaza Ho, trabajar en el departamento de soporte incrementa en un 8% el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

dep_ventas, No se rechaza Ho, trabajar en el departamento de ventas no afecta el riesgo de renunciar con respecto a trabajar en el departamento de investigación.

evaluacion, No se rechaza H0, la evalaución no afecta el riesgo de que un empleado renuncie.

promedio_horas_mes, No se rechaza H0, el número de horas trabajadas no afecta el riesgo de que un empleado renuncie.

proyectos, se rechaza H0, el número de proyectos reduce en un 10% el riesgo de que un empleado renuncie.

salario bajo, Se rechaza H0, recibir un salario bajo incrementa en un 41% el riesgo de que un empleado renuncie con respecto a los que ganan un salrio alto.

salario medio, Se rechaza H0, recibir un salario medio incrementa el riesgo de que un empleado renuncie en un 25% con respecto a los que ganan un salario alto.

satisfaccion, se rechaza H0, que un empleado este satisfecho reduce en un 7% el riesgo de que renuncie de la empresa.